### PR TITLE
uri comparison fix, adding redirect_uri_validation_failed msal error

### PR DESCRIPTION
**Pull request by SomkaPe**
Add support for default redirect uri in iOS and Android https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/391

### CODE REVIEW

**Comment by kpanwar** [comment]
This is not the fix we want. Please see latest issue assigned to you.

File: src/Microsoft.Identity.Client/Platforms/iOS/PlatformInformation.cs [hunk 1]
```diff
@@ -73,8 +73,8 @@ public override void ValidateRedirectUri(Uri redirectUri, RequestContext request
         {
             base.ValidateRedirectUri(redirectUri, requestContext);
 
-            if (PublicClientApplication.DEFAULT_REDIRECT_URI.Equals(redirectUri))
-                throw new MsalException("Default redirect URI - " + PublicClientApplication.DEFAULT_REDIRECT_URI +
+            if (PublicClientApplication.DEFAULT_REDIRECT_URI.Equals(redirectUri.AbsoluteUri))
```
Comment (kpanwar) [inline src/Microsoft.Identity.Client/Platforms/iOS/PlatformInformation.cs]: see bug to make correct fix

**Comment by SomkaPe** [comment]
updated PR  according to https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/391

File: src/Microsoft.Identity.Client/Platforms/Android/PlatformInformation.cs [hunk 2]
```diff
@@ -35,6 +35,8 @@ namespace Microsoft.Identity.Client
     [Android.Runtime.Preserve(AllMembers = true)]
     internal class PlatformInformation : PlatformInformationBase
     {
+        internal const string IosDefaultRedirectUri = "msal{0}://auth";
```
Comment (kpanwar) [inline src/Microsoft.Identity.Client/Platforms/Android/PlatformInformation.cs]: rename to redirectUriTemplate.

File: src/Microsoft.Identity.Client/Platforms/iOS/PlatformInformation.cs [hunk 3]
```diff
@@ -35,6 +35,8 @@ namespace Microsoft.Identity.Client
 {
     internal class PlatformInformation : PlatformInformationBase
     {
+        internal const string IosDefaultRedirectUri = "msal{0}://auth";
```
Comment (kpanwar) [inline src/Microsoft.Identity.Client/Platforms/iOS/PlatformInformation.cs]: rename as above

**Comment by kpanwar** [comment]
@SomkaPe - your change is very well thought out. I really like it. I wish I could have thought of this refactor.

File: src/Microsoft.Identity.Client/Platforms/Android/PlatformInformation.cs (hunk 2 above)
Comment (SomkaPe) [inline src/Microsoft.Identity.Client/Platforms/Android/PlatformInformation.cs]: done

File: src/Microsoft.Identity.Client/Platforms/iOS/PlatformInformation.cs (hunk 3 above)
Comment (SomkaPe) [inline src/Microsoft.Identity.Client/Platforms/iOS/PlatformInformation.cs]: done

**Pull request merged**